### PR TITLE
Add feature flag for opening in new tab

### PR DESCRIPTION
**Pull request by CrisBarreiro**
Task/Issue URL: https://app.asana.com/0/0/1208511584623404/f 

### Steps to test this PR

_Feature 1_
- [x] Check that with the prod RC, the setting to open duck player in new tab is not available and Duck Player is opened in the same page

_Feature 2_
- [x] Go to FF inventory and turn DuckPlayer > newtab on
- [x] Check the setting to open Duck Player in new tab is now visible in Duck Player settings and works as expected 

### UI changes
| Before  | After |
| ------ | ----- |
!(Upload before screenshot)|(Upload after screenshot)|

### CODE REVIEW

**Comment by CrisBarreiro** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5124?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#5125** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5125?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5124** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5124?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#5121** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5121?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5109** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5109?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#5110](https://github.com/duckduckgo/Android/pull/5110) <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5110?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CrisBarreiro and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: privacy-config/privacy-config-api/src/main/java/com/duckduckgo/privacy/config/api/PrivacyFeatureName.kt [hunk 1]
```diff
@@ -27,4 +27,4 @@ enum class PrivacyFeatureName(val value: String) {
     TrackingParametersFeatureName("trackingParameters"),
 }
 
-const val PRIVACY_REMOTE_CONFIG_URL = "https://staticcdn.duckduckgo.com/trackerblocking/config/v4/android-config.json"
+const val PRIVACY_REMOTE_CONFIG_URL = "https://www.jsonblob.com/api/1252645298605252608"
```
Comment (joshliebe) [inline privacy-config/privacy-config-api/src/main/java/com/duckduckgo/privacy/config/api/PrivacyFeatureName.kt:30]: You probably don't want to change this 😁

File: duckplayer/duckplayer-impl/src/main/java/com/duckduckgo/duckplayer/impl/DuckPlayerDataStore.kt [hunk 2]
```diff
@@ -194,7 +194,7 @@ class SharedPreferencesDuckPlayerDataStore @Inject constructor(
     private val duckPlayerOpenInNewTab: Flow<Boolean>
         get() = store.data
             .map { prefs ->
-                prefs[Keys.DUCK_PLAYER_OPEN_IN_NEW_TAB] ?: false
+                prefs[Keys.DUCK_PLAYER_OPEN_IN_NEW_TAB] ?: true
```
Comment (joshliebe) [inline duckplayer/duckplayer-impl/src/main/java/com/duckduckgo/duckplayer/impl/DuckPlayerDataStore.kt:197]: nit: What’s the reason for defaulting to `true` here when the toggle defaults to `false`?

File: duckplayer/duckplayer-impl/src/main/java/com/duckduckgo/duckplayer/impl/RealDuckPlayer.kt [hunk 3]
```diff
@@ -434,11 +439,16 @@ class RealDuckPlayer @Inject constructor(
...
+        if (!duckPlayerFeature.openInNewTab().isEnabled()) return Unavailable
+        return if (duckPlayerFeatureRepository.shouldOpenInNewTab()) On else Off
     }
 
-    override fun observeShouldOpenInNewTab(): Flow<Boolean> {
-        return duckPlayerFeatureRepository.observeOpenInNewTab()
+    override fun observeShouldOpenInNewTab(): Flow<OpenDuckPlayerInNewTab> {
+        return duckPlayerFeatureRepository.observeOpenInNewTab().map {
+            (if (!duckPlayerFeature.openInNewTab().isEnabled()) Unavailable else if (it) On else Off).also {
+                Log.d("Cris", "observeShouldOpenInNewTab: $it")
```
Comment (joshliebe) [inline duckplayer/duckplayer-impl/src/main/java/com/duckduckgo/duckplayer/impl/RealDuckPlayer.kt:450]:
```suggestion
                Log.d(“RealDuckPlayer", "observeShouldOpenInNewTab: $it")
```

Comment (joshliebe) [inline duckplayer/duckplayer-impl/src/main/java/com/duckduckgo/duckplayer/impl/RealDuckPlayer.kt:450]: Remove this Log or use Timber instead.

**Review by joshliebe — CHANGES_REQUESTED** [review]
Works as expected but some debugging code slipped in. Remove that and I’ll approve 🙂

File: duckplayer/duckplayer-impl/src/main/java/com/duckduckgo/duckplayer/impl/RealDuckPlayer.kt [hunk 4]
```diff
@@ -433,11 +437,14 @@ class RealDuckPlayer @Inject constructor(
...
+    override suspend fun shouldOpenDuckPlayerInNewTab(): OpenDuckPlayerInNewTab {
+        if (!duckPlayerFeature.openInNewTab().isEnabled()) return Unavailable
+        return if (duckPlayerFeatureRepository.shouldOpenInNewTab()) On else Off
     }
 
-    override fun observeShouldOpenInNewTab(): Flow<Boolean> {
-        return duckPlayerFeatureRepository.observeOpenInNewTab()
+    override fun observeShouldOpenInNewTab(): Flow<OpenDuckPlayerInNewTab> {
+        return duckPlayerFeatureRepository.observeOpenInNewTab().map {
+            (if (!duckPlayerFeature.openInNewTab().isEnabled()) Unavailable else if (it) On else Off)
```
Comment (joshliebe) [inline duckplayer/duckplayer-impl/src/main/java/com/duckduckgo/duckplayer/impl/RealDuckPlayer.kt:447]: The enclosing parenthesis aren’t needed here anymore

**Review by joshliebe — APPROVED** [review]
Just the one [comment](https://github.com/duckduckgo/Android/pull/5124#discussion_r1804421660), LGTM!

**Comment by CrisBarreiro** [comment]
### Merge activity

* **Oct 17, 2:03 PM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/5124).

**Pull request merged**